### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by ramaneswaran**
Made the code functional and stateless
Added better Date extraction
Added function for graceful failure
Added inline submenu to  edit collected details
The bot is concurrent and threaded
Using a generator to iterate through messages, so messages not exposed

### CODE REVIEW

File: src/bot/con.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{'game_short_name': None, 
+'chat_instance': '-2248192099587011709', 
+'id': '2469572960604862310',
```
Comment (Geek-ubaid) [inline src/bot/con.py]: why is this hardcoded?

File: src/bot/con.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{'game_short_name': None, 
+'chat_instance': '-2248192099587011709', 
+'id': '2469572960604862310', 
+'from_user': {'id': 574992261, 'is_bot': False, 'first_name': 'Raman', 'username': None, 'last_name': None,
```
Comment (Geek-ubaid) [inline src/bot/con.py]: yeah here also id is hardcoded!

File: src/bot/requirements.txt [hunk 3]
```diff
@@ -1,8 +1,192 @@
...
+uvloop==0.14.0
+wasabi==0.6.0
+wcwidth==0.1.9
+webencodings==0.5.1
+webexteamssdk==1.1.1
+websockets==8.1
+Werkzeug==1.0.1
+wrapt==1.12.1
+yarl==1.4.2
+zipp==3.1.0
```
Comment (Geek-ubaid) [inline src/bot/requirements.txt]: Lot of dependencies, revise the requirements list, use pipreqs to generate the requirements.txt based on imports in the your project!

**Review by Geek-ubaid — CHANGES_REQUESTED** [review]

File: src/bot/con.py (hunk 2 above)
Comment (ramaneswaran) [reply src/bot/con.py]: That's just an example response 
I use that as reference

File: src/bot/requirements.txt (hunk 3 above)
Comment (ramaneswaran) [reply src/bot/requirements.txt]: Okay , will make the changes

**Review by Geek-ubaid — APPROVED** [review]

**Pull request merged**